src/lightning/pytorch/callbacks/spike.py

```diff
@@ -0,0 +1,31 @@
+import os
+from typing import Any, Mapping, Union
+
+import torch
+
+import lightning.pytorch as pl
+from lightning.fabric.utilities.spike import SpikeDetection as FabricSpikeDetection
+from lightning.pytorch.callbacks.callback import Callback
+
+
+class SpikeDetection(FabricSpikeDetection, Callback):
+    @torch.no_grad()
+    def on_train_batch_end(  # type: ignore
+        self,
+        trainer: pl.Trainer,
```

Comment:
```suggestion
        trainer: "pl.Trainer",
```
helps with circular imports hopefully